src/Acf.php

```diff
@@ -185,4 +199,23 @@ private static function apply_default_transform( $field_obj ) {
 	private static function get_transform_class_name( $field_type ) {
 		return '\\' . __NAMESPACE__ . '\\Acf\\Transforms\\' . str_replace( '_', '', ucwords( $field_type, '_' ) );
 	}
+
+	/**
+	 * Get an ACF group
+	 *
+	 * @param $group_id
+	 * @return bool
+	 */
+	private static function get_group( $group_id ) {
+		// @codingStandardsIgnoreStart
+		// Ignore use WP_Query rule because we don't know if this will be a sub-query and hence create complications.
```

Comment:
I had a look at get_page_by_path - https://core.trac.wordpress.org/browser/tags/4.4.2/src/wp-includes/post.php#L4084 - it's running an extra query first to get the id form the post_name, then it gets the post using get_post. So it must be less efficient than get_posts which presumably runs just 1 query.

Also I don't think this was the intended use of the function - well I think it's meant for getting a pst from the url, rather than the post_name directly.

What do you think?